File: Beecrowd/6-Paradigmas/1034-Festival_de_Estátuas_de_Gelo-8/1034.cs
using System;

class Program {
    static void Main() {
        int T = int.Parse(Console.ReadLine());

        while (T-- > 0) {
            string[] valores = Console.ReadLine().Split();
            int N = int.Parse(valores[0]);
            int M = int.Parse(valores[1]);

            int[] blocos = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
            int[] dp = new int[M + 1];

            Array.Fill(dp, int.MaxValue);
            dp[0] = 0;

            foreach (int bloco in blocos) {
                for (int j = bloco; j <= M; j++) {
                    if (dp[j - bloco] != int.MaxValue) {
                        dp[j] = Math.Min(dp[j], dp[j - bloco] + 1);
                    }
                }
            }

            Console.WriteLine(dp[M]);
        }
    }
}

File: Beecrowd/6-Paradigmas/1034-Festival_de_Estátuas_de_Gelo-8/1034.java
import java.util.Scanner;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        
        while (T-- > 0) {
            int N = sc.nextInt();
            int M = sc.nextInt();
            
            int[] blocos = new int[N];
            for (int i = 0; i < N; i++) {
                blocos[i] = sc.nextInt();
            }
            
            int[] dp = new int[M + 1];
            Arrays.fill(dp, Integer.MAX_VALUE);
            dp[0] = 0;

            for (int bloco : blocos) {
                for (int j = bloco; j <= M; j++) {
                    if (dp[j - bloco] != Integer.MAX_VALUE) {
                        dp[j] = Math.min(dp[j], dp[j - bloco] + 1);
                    }
                }
            }

            System.out.println(dp[M]);
        }
        
        sc.close();
    }
}
